js/stats.js

```diff
@@ -1,173 +1,222 @@
+
+    Note: Unless otherwise noted, all code henceforth in this file
+        has been authored by me, Tony J Huang. 
+*/
+
+/*
+    Global export for _$_ variable. 
+
+    Sample usages include: 
+      _$_.sSplit(...);
+      _$_.stDev(...);
+*/
+var _$_ = (function() {
+    var my = {};
+
+
+    // Given a object, checks if it has a value at fieldName.
+    // Otherwise throws an error.
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    function ensure(params, fieldName) {
+        if(params[fieldName])
+            return params[fieldName];
+        throw new Error("missing param field: " + fieldName);
+    }
+
+
+    // Validates the input as a properly formatted Array of Pictures.
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    function validatePictures(objects) {
+        if(!_.isArray(pictures))
+            throw new Error(
+                "Pictures argument is not an Array of Picture");
+
+        if(!_.every(objects, o => o instanceof Picture))
+            throw new Error(
+                "Pictures argument is not properly formatted, ie: " + o);
+    }
+
+
+    // Validates the other user defined input params ansd makes sure
+    // they make sense/are possible.
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    function validateNumArgs(numGroups, numPictures, pictures) {
+        var size = _.size(pictures);        
+        if(numGroups * numPictures > size)
+            throw new Error("Can't split " + size + " total pictures into " 
+                + numGroups + "groups of " + numPictures);
+    }
+
+
+    // Chunks an array into equal sizes of chunkSize, notwithstanding 
+    // the very last chunk, which is anywhere from size 1 to chunkSize
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    Array.prototype.chunk = function(chunkSize) {
+        var R = [];
+        for (var i = 0; i < this.length; i += chunkSize)
+            R.push(this.slice(i, i + chunkSize));
+        return R;
+    }
+
+
+    // Sorts pictures by their rating distance from targetRating.
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    function sortPicturesByRating(targetRating, pictures) {
+        return _.sortBy(pictures, p => Math.abs(p.rating - targetRating));
+    }
+
+
+    
+    // Round Robin.
+    // ALGORITHM: Picks the picture with the closest rating to the target
+    //     and assigns it to group 1, then group 2, then group 3, ...,
+    //     until all groups are full.
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    
+    function splitRR(numGroups, numPictures, targetRating, pictures) {
+        var sortedPictures = sortPicturesByRating(targetRating, pictures);
+        var candidates = _.first(sortedPictures, numGroups * numPictures);
+        var i = 0;
+        return _.values(_.groupBy(candidates), p => i++ % numGroups);
```

Comment:
not sure if arrow functions are supported in air - this is only supported in ecmascript 6 (new javascript)